tests/nodeos_read_terminate_at_block_test.py

```diff
@@ -0,0 +1,316 @@
+   time.sleep(1) # Give a second to replay or resync if needed
+   assert isRelaunchSuccess, relaunchAssertMessage
+   return isRelaunchSuccess
+
+# List to contain the test result message
+testResultMsgs = []
+testSuccessful = True
+try:
+   # Kill any existing instances and launch cluster
+   TestHelper.printSystemInfo("BEGIN")
+   cluster.killall(allInstances=killAll)
+   cluster.cleanup()
+   cluster.launch(
+      prodCount=numOfProducers,
+      totalProducers=numOfProducers,
+      totalNodes=totalNodes,
+      pnodes=1,
+      useBiosBootFile=False,
+      topo="mesh",
+      specificExtraNodeosArgs={
+         0:"--enable-stale-production",
+         4:"--read-mode irreversible",
+         6:"--read-mode irreversible",
+         9:"--plugin eosio::producer_api_plugin"})
+
+   producingNodeId = 0
+   producingNode = cluster.getNode(producingNodeId)
+
+   def stopProdNode():
+      if not producingNode.killed:
+         producingNode.kill(signal.SIGTERM)
+
+   def startProdNode():
+      if producingNode.killed:
+         relaunchNode(producingNode)
+
+   # Give some time for it to produce, so lib is advancing
+   waitForBlksProducedAndLibAdvanced()
+
+   # Kill all nodes, so we can test all node in isolated environment
+   for clusterNode in cluster.nodes:
+      clusterNode.kill(signal.SIGTERM)
+   cluster.biosNode.kill(signal.SIGTERM)
+
+   # Wrapper function to execute test
+   # This wrapper function will resurrect the node to be tested, and shut it down by the end of the test
+   def executeTest(nodeIdOfNodeToTest, runTestScenario):
+      testResult = False
+      resultDesc = None
+      try:
+         # Relaunch killed node so it can be used for the test
+         nodeToTest = cluster.getNode(nodeIdOfNodeToTest)
+         Utils.Print("Re-launch node #{} to excute test scenario: {}".format(nodeIdOfNodeToTest, runTestScenario.__name__))
+         relaunchNode(nodeToTest, relaunchAssertMessage="Fail to relaunch before running test scenario")
+
+         # Run test scenario
+         runTestScenario(nodeIdOfNodeToTest, nodeToTest)
+         resultDesc = "!!!TEST CASE #{} ({}) IS SUCCESSFUL".format(nodeIdOfNodeToTest, runTestScenario.__name__)
+         testResult = True
+      except Exception as e:
+         resultDesc = "!!!BUG IS CONFIRMED ON TEST CASE #{} ({}): {}".format(nodeIdOfNodeToTest, runTestScenario.__name__, e)
+      finally:
+         Utils.Print(resultDesc)
+         testResultMsgs.append(resultDesc)
+         # Kill node after use
+         if not nodeToTest.killed: nodeToTest.kill(signal.SIGTERM)
+      return testResult
+
+
+   # 1th test case: Replay in irreversible mode with reversible blks while connected to producing node
+   # Expectation: Node replays and launches successfully
+   #              and the head and lib should be no less than termAtBlockNum
+   def replayInIrrModeWithRevBlksAndConnectedToProdNodeTermAtBlock(nodeIdOfNodeToTest, nodeToTest):
```

Comment:
I don't see `nodeIdOfNodeToTest` is used somewhere in the function but even if it does, there is a property for this:
`Node.nodeId`